sync/src/block/downloader/header.rs

```diff
@@ -143,7 +148,7 @@ impl HeaderDownloader {
         if self.best_hash == self.pivot.hash {
             ctrace!(SYNC, "Ignore received headers, pivot already reached the best hash");
         } else if first_header_hash == self.pivot.hash {
-            for header in headers.iter() {
+            for header in headers[1..].iter() {
```

Comment:
Would you explain it?

Reply:
Currently, the responded header data always makes one overlap. I made `queued` to reduce re-importing already queued headers but previous routine still generates one overlap every 128 requested headers. So I decided to iterate from the index 1.

Reply:
@Kais-DkM Could you explain why the response overlaps one header?

Reply:
You can see my comment [here](https://github.com/CodeChain-io/codechain/issues/1511#issuecomment-495173807).
In short, no specific reason for that.

Reply:
@HoOngEe I think it's better to keep the original code.
If you really want to improve the performance of it, you should make the response exclude the overlapped one, not just skipping the first one.

Reply:
@sgkim126 Ok, I reverted it.